components/async-container/async-container-mixin.js

```diff
@@ -0,0 +1,78 @@
+
+export const asyncStates = {
+	initial: 'initial',
+	pending: 'pending',
+	success: 'success',
+	failure: 'failure'
+};
+
+export const AsyncContainerMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			asyncState: { type: String }
```

Comment:
In [Justin's example](https://youtu.be/x9YDQUJx2uw), he tracks both whether or not there are pending children in addition to the actual count. That means that a change in either could cause a re-render... which I don't love since for a lot of cases you don't care how many there are.

Just wanted to make sure that knowing the number of pending children isn't something we foresee needing?

Reply:
Yeah I think I saw that in his code and for the moment decided to keep the counts internal, but at least having them internal is useful for determining the state.  We can easily expose if needed.